File: L151.py
def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        n=len(s)
        i=0
        res=""
        while i<n:
            while i<n and s[i]==" ":
                i+=1
            if i>=n:
                break
            j=i+1
            while j<n and s[j]!=" ":
                j+=1
            res=s[i:j]+" "+res
            i=j+1
        return res.strip()

File: L151.java
public class L151 {

    public String reverseWords(String s) {
        int n = s.length();
        int j = n - 1;
        StringBuilder bro = new StringBuilder();

        while (j >= 0) {
            // skip trailing spaces
            while (j >= 0 && s.charAt(j) == ' ')
                j--;

            if (j < 0)
                break;

            int end = j;

            // move j left until space or start
            while (j >= 0 && s.charAt(j) != ' ')
                j--;

            // extract the word
            if (bro.length() > 0)
                bro.append(" ");
            bro.append(s.substring(j + 1, end + 1));
        }

        return bro.toString();
    }

    //see this also
    public String reverseWord(String s) {
        String[] words = s.split("\\s+");
        StringBuilder res = new StringBuilder();

        for (int i = words.length - 1; i >= 0; i--) {
            res.append(words[i]);
            if (i != 0) {
                res.append(" ");
            }
        }

        return res.toString().trim();
    }
}